### DISCUSSION TITLE
Generic OAuth2 - new user overwrites current user.

### DISCUSSION THREAD

**Discussion by Dental1347** [requarks/wiki#7098] [Help / Questions]
Hello,

Recently started using wiki.js. I make use of authentik, and have configured Generic OAuth2 within the authentication tab. I've been facing two different issues thus far:
1. When having auto-register disabled, users are still able to login when they login with the Generic OAuth2 account via authentik. I did not create these users, nor should they be allowed to login if I understand correctly?
2. This is by far the weirdest one, and I have no clue what is causing this. When I have auto register on (or off for that matter) users are able to login, and user ID 3 is overwritten by the last login. Meaning, if I'm logged into the wiki, as an administrator for account ID 3, when another authentik user authenticates, they will overwrite my user and become administrator themselves. I first thought this was behavior of not enabling auto register. But when enabling this, the user was still overwritten, and the created group for new users was ignored and they still became admin. 
3. When setting the variable displayName, the name of the account is overwritten on login to `???`. Even if I manually change the displyname, it is overwritten on logon by ???. I changed this to name, which seems to correctly pull my display name from authentik.

Could you guys clarify if I'm perhaps setting things up incorrectly, or if we're dealing with a nasty bug? (probably the first :))

Screenshots of configuration.
![firefox_Zfe8FXu50p](https://github.com/requarks/wiki/assets/44779473/fc798c4d-63ab-4c5d-b30e-4fa44f76fdb9)
![firefox_YheAgQHtsm](https://github.com/requarks/wiki/assets/44779473/3e9780ba-2769-4a36-bcaf-6eeb8253335d)
User being overwritten on every login:
![firefox_dntJIPVCUD](https://github.com/requarks/wiki/assets/44779473/4603ac55-258d-4aa4-865e-96b55c6d913e)

Help is much appreciated, because I can't open up the wiki when stuff like this is happening.

**Comment by Dental1347** [comment]
What I want to achieve:

Use Generic OAuth2 and make it so auto register is off, so users cannot create an account (unless I turn it on).
Have it not overwrite every user, but NEW users.
Being able to properly fetch the userName (unless name is just fine and not causing issues)

The goal is a private wiki, I have removed the local (or hidden) login possibility, authentication is redirected to authentik subdomain, guest access has been removed.

**Comment by mus65** [comment]
Point 2 is probably caused by server-side caching, see https://github.com/requarks/wiki/discussions/6822#discussioncomment-9533304

**Comment by ptrstr** [comment]
I had a similar issue. Make sure the provider ID is actually distinct and valid for every user
